### PR TITLE
Avoid direct access to Args/Params

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32243

## Description
added a validation that checks for args/params if a direct access was added.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py [hunk 1]
```diff
@@ -28,11 +33,17 @@ class XsoarChecker(BaseChecker):
     def __init__(self, linter=None):
         super(XsoarChecker, self).__init__(linter)
         self.is_script = True if os.getenv('is_script') == 'True' else False
+        self.common_args_params = ['args', 'dargs', 'arguments', 'd_args', 'data_args', 'params', 'PARAMS',
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py:36]: are there many usages of `'params', 'PARAMS'`. if not too many - i'm ok with that creating a warning.
```suggestion
        self.common_args_params = ['args', 'dargs', 'arguments', 'd_args', 'data_args', 'params', 'PARAMS',
```

File: demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py [hunk 2]
```diff
@@ -63,6 +74,21 @@ def _not_implemented_error_in_main(self, node):
...
+    def _direct_access_dict_checker(self, node):
+        try:
+            # for demisto.args()[] implementation or for demisto.params()[]
+            if node.value.func.expr.name == 'demisto' and node.value.func.attrname == 'args':
+                self.add_message("direct-access-dict-exist", node=node)
+            elif node.value.func.expr.name == 'demisto' and node.value.func.attrname == 'params':
+                self.add_message("direct-access-dict-exist", node=node)
+        except Exception:
+            try:
+                # for args[] implementation
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/resources/pylint_plugins/xsoar_level_checker.py:86]:
```suggestion
                # for args[]/params[] implementation
```

**Review by bakatzir — COMMENTED** [review]
can you add a circle build of a run on the content repo?

**Comment by orhovy** [comment]
https://app.circleci.com/pipelines/github/demisto/content/58143/workflows/6e3e040b-2dfe-4376-a4a4-fd0512048367
build on content

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
 * Added an option for **format** to run on a whole pack.
 * Added new validation of unimplemented commands from yml in the code to `XSOAR-linter`.
 * Fixed an issue where Auto-Extract fields were only checked for newly added incident types in the **validate** command.
+* Added new validation of direct access to args/params dicts to `XSOAR-linter`.
```
Comment (bakatzir) [inline CHANGELOG.md:7]:
```suggestion
* Added a new warning validation of direct access to args/params dicts to `XSOAR-linter`.
```

**Review by bakatzir — COMMENTED** [review]
please consider adding a test case when making the param/args is in the right side,
`test_direct_access_exists_as_var` - in this case as we said no warning is expected.

**Comment by orhovy** [comment]
@bakatzir such test already exist, I have added it to the warning file which xsoar_integration_tests run on , and check for the warning.

**Review by bakatzir — APPROVED** [review]

**Review by roysagi — COMMENTED** [review]
@orhovy - please run lint -a to check this does not break anything.

**Comment by orhovy** [comment]
@roysagi , ran it already in content repo on all. attached the build with all the warnings in the early comments.
It didnt break anything as its a warning message.

**Pull request merged**